### PR TITLE
Bump Ubuntu to 24.04

### PR DESCRIPTION
**Pull request by vrabaud**

### CODE REVIEW

File: .github/workflows/ci-disable-gtest.yml [hunk 1]
```diff
@@ -28,11 +28,11 @@ jobs:
...
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-24.04]
         include:
-          - runs-on: ubuntu-latest
+          - runs-on: ubuntu-24.04
             compiler: gcc
-            gcc: 12
+            gcc: 13
```
[the remark this answers is not in the document]
Comment (vrabaud) [reply .github/workflows/ci-disable-gtest.yml:35]: I thought that was only through a ppa, let's see.

File: .github/workflows/ci-linux-static-old-local.yml [hunk 2]
```diff
@@ -40,6 +40,11 @@ jobs:
         profile: minimal
         toolchain: stable
         override: true
+    - name: Install dependencies (Linux)
+      if: runner.os == 'Linux'
```
Comment (wantehchang) [inline .github/workflows/ci-linux-static-old-local.yml:44]: Nit: The Linux test can be omitted. See line 48 and line 51.

File: .github/workflows/ci-unix-static.yml [hunk 3]
```diff
@@ -42,11 +42,16 @@ jobs:
...
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt update
+        sudo apt install imagemagick libjpeg-turbo8-dev libpng-dev 
     - name: Install dependencies (macOS)
       if: runner.os == 'macOS'
       run: brew install imagemagick
     - name: Install cargo-c (linux)
-      if: matrix.os == 'ubuntu-latest'
+      if: matrix.os == 'ubuntu-24.04'
```
Comment (wantehchang) [inline .github/workflows/ci-unix-static.yml:54]: Nit: we can test `runner.os == 'Linux'` instead.

File: .github/workflows/ci-unix-static.yml [hunk 4]
```diff
@@ -68,7 +73,7 @@ jobs:
         path: ext, build/_deps
         key: ${{ runner.os }}-${{ matrix.build-type }}-unix-static-${{ hashFiles('cmake/Modules/*', 'ext/*.cmd', 'ext/*.sh') }}
     - name: Setup cmake
-      if: matrix.os == 'ubuntu-latest'
+      if: matrix.os == 'ubuntu-24.04'
```
Comment (wantehchang) [inline .github/workflows/ci-unix-static.yml:76]: Nit: I think we can test `runner.os == 'Linux'` instead.

**Review by wantehchang — APPROVED** [review]

**Review by wantehchang — COMMENTED** [review]
The commit message should have a body that describes the other changes in this pull request, such as,

- Bump gcc version to 14
- Install the imagemagick, libjpeg-turbu8-dev, and libpng-dev packages

File: .github/workflows/ci-linux-static-old-local.yml (hunk 2 above)
Comment (vrabaud) [reply .github/workflows/ci-linux-static-old-local.yml:44]: Done

File: .github/workflows/ci-unix-static.yml (hunk 3 above)
Comment (vrabaud) [reply .github/workflows/ci-unix-static.yml:54]: Done

File: .github/workflows/ci-unix-static.yml (hunk 4 above)
Comment (vrabaud) [reply .github/workflows/ci-unix-static.yml:76]: Done

**Pull request merged**